tornado-unittests/src/main/java/uk/ac/manchester/tornado/unittests/compute/LLMFusedKernelsTest.java

```diff
@@ -0,0 +1,286 @@
+     * limitations under the License.
+     *
+     */
+    package uk.ac.manchester.tornado.unittests.compute;
+
+    import org.junit.Test;
+    import uk.ac.manchester.tornado.api.GridScheduler;
+    import uk.ac.manchester.tornado.api.ImmutableTaskGraph;
+    import uk.ac.manchester.tornado.api.KernelContext;
+    import uk.ac.manchester.tornado.api.TaskGraph;
+    import uk.ac.manchester.tornado.api.TornadoExecutionPlan;
+    import uk.ac.manchester.tornado.api.WorkerGrid;
+    import uk.ac.manchester.tornado.api.WorkerGrid1D;
+    import uk.ac.manchester.tornado.api.enums.DataTransferMode;
+    import uk.ac.manchester.tornado.api.exceptions.TornadoExecutionPlanException;
+    import uk.ac.manchester.tornado.api.math.TornadoMath;
+    import uk.ac.manchester.tornado.api.types.arrays.FloatArray;
+    import uk.ac.manchester.tornado.unittests.common.TornadoTestBase;
+
+    import java.util.stream.IntStream;
+
+    import static org.junit.Assert.assertEquals;
+
+
+    /**
+     * Test to validate the functionality of benchmark compute layers or components of a Large Language Model (LLM) transformer,
+     * focusing on operations like RMS Normalization (RNSNorm) and Matrix-Vector Multiplication (MatMul).
+     *
+     * The test verifies that the fused kernel approach, which combines RMSNorm and MatMul in a single execution flow,
+     * produces correct results by comparing the output of a parallel (fused) execution with a sequential (reference) implementation.
+     *
+     * Specifically, the test checks:
+     * 1. **RMS Normalization (RNSNorm)**:
+     *    - RMSNorm is computed by normalizing input values using the inverse square root of the sum of squares, scaled by a weight factor.
+     *    - The test compares the result of this normalization from the parallel execution against a sequential computation.
+     *
+     * 2. **Matrix-Vector Multiplication (MatMul)**:
+     *    - The test also ensures the matrix-vector multiplication operation (MatMul) works correctly when combined with the RMSNorm step, ensuring that the fusion of these operations does not compromise their correctness.
+     *
+     * The execution is tested in two forms:
+     * - **Sequential Execution**: Using traditional CPU-based loops for reference results.
+     * - **Fused Parallel Execution**: Leveraging TornadoVM's parallel computing capabilities to execute the same operations using fused kernels.
+     *
+     * For both the RMSNorm and MatMul operations, the test compares the result from the parallel execution with the sequential reference to ensure that the fused parallel approach produces identical outputs, within an acceptable error tolerance.
+     *
+     * <p>
+     * How to run the tests?
+     * </p>
+     *
+     * <code>
+     * tornado-test -V uk.ac.manchester.tornado.unittests.compute.LLMFusedKernelsTest
+     * </code>
+     */
+
+    public class LLMFusedKernelsTest extends TornadoTestBase {
```

Comment:
To keep consistency with other test classes, I would suggest to move the "Test" at the beginning of the name of the class.